### PR TITLE
Fix A Bunch Of Memory Leaks

### PR DESCRIPTION
**Pull request by pamaddox**

### CODE REVIEW

File: src/modules/check_test.c [hunk 1]
```diff
@@ -204,8 +204,12 @@ rest_test_check_result(struct check_test_closure *cl) {
 
   mtevL(nlerr, "Flushing check test result\n");
 
-  if(cl->restc->call_closure_free)
+  if(cl->restc->call_closure_free) {
     cl->restc->call_closure_free(cl->restc->call_closure);
+    /* We have to free this manually because the closure set by
+     * rest_get_xml_upload doesn't do it */
+    free(cl->restc->call_closure);
```
Comment (aidancully) [inline src/modules/check_test.c]: seems we should fix `rest_get_xml_upload`, instead?

Comment (pamaddox) [reply src/modules/check_test.c]: I can look into that, but it's riskier. Lots of other stuff depends on that function and it's an external library call.

Comment (aidancully) [reply src/modules/check_test.c]: It's my strong opinion that we should fix the free function set by `rest_get_xml_upload`... The abstraction boundary is such that anything that sets the `call_closure` should be the thing that knows how to free the `call_closure`.

Comment (pamaddox) [reply src/modules/check_test.c]: OK, fixed in libmtev and reverted calls to manually fix it here.

**Pull request merged**